content-tools/update-cdids/main.go

```diff
@@ -0,0 +1,486 @@
+	PublishDate     interface{} `json:"publishDate"`
+	Teams           []string    `json:"teams"`
+	CollectionOwner string      `json:"collectionOwner"`
+	ReleaseURI      interface{} `json:"releaseUri"`
+	IsEncrypted     bool        `json:"isEncrypted"`
+}
+
+type Credential struct {
+	Email    string `json:"email"`
+	Password string `json:"password"`
+}
+
+type CollectionDescriptionResponse struct {
+	ID string `json:"id"`
+}
+
+type CdIDPair struct {
+	cdID    string
+	oldCdID string
+}
+
+func (cp *CdIDPair) Print(ctx context.Context, prefix string) {
+	log.Event(ctx, fmt.Sprintf("%s cdid: %s, oldcdid: %s", prefix, cp.cdID, cp.oldCdID), log.INFO)
+}
+
+func main() {
+	setupFlags()
+
+	ctx := context.Background()
+	httpClient := http.DefaultClient
+
+	if validateMandatoryParams(ctx) {
+		return
+	}
+
+	cdIDPairs := readCdIDPairs(ctx, filePath, sheetname, limit)
+	for _, pair := range cdIDPairs {
+		pair.Print(ctx, "Parsed ")
+	}
+	log.Event(ctx, fmt.Sprintf("Read %d cdIDPairs", len(cdIDPairs)), log.INFO)
+	ctx, err := authenticate(ctx, httpClient, environment, username, password)
+	if err != nil {
+		errMessage := fmt.Errorf("error occurred while authenticating. Stopping the script. error: %s", err.Error())
+		log.Error(errMessage)
+		return
+	}
+
+	collectionID, err := createCollection(ctx, httpClient, getCollectionName(), environment)
+	if err != nil {
+		errMessage := fmt.Errorf("error occurred while creating collection. Stopping the script. error: %s", err.Error())
+		log.Error(errMessage)
+		return
+	}
+	log.Event(ctx, fmt.Sprintf("created collection: %s", collectionID), log.INFO)
+
+	for _, pair := range cdIDPairs {
+		pair.Print(ctx, "Processing ")
+
+		oldCdIDLocation, err := searchOldCdID(ctx, httpClient, environment, pair.oldCdID)
+		if err != nil {
+			pair.Print(ctx, "stopping. Error occurred while fetching cdID location")
+			continue
+		}
+
+		existingCollectionID, err := checkIfCdIDExistsInAnotherCollection(ctx, httpClient, environment, oldCdIDLocation)
+		if err != nil {
+			pair.Print(ctx, "stopping. Error occurred while verifying if cdID location exists in another collection")
+			continue
+		}
+
+		if len(existingCollectionID) > 0 {
+			pair.Print(ctx, fmt.Sprintf("stopping. Error occurred. CDID: %s already exists in another collection: %s", pair.oldCdID, existingCollectionID))
+			continue
+		}
+
+		cdIDData, err := fetchDataForCDID(ctx, httpClient, environment, oldCdIDLocation)
+		if err != nil {
+			pair.Print(ctx, "stopping. Error occurred while fetching cdID data.")
+			continue
+		}
+
+		err = addCDIDToCollection(ctx, httpClient, environment, collectionID, oldCdIDLocation, cdIDData)
+		if err != nil {
+			pair.Print(ctx, "stopping. Error occurred while adding CDID to collection")
+			continue
+		}
+
+		err = approveCDID(ctx, httpClient, environment, collectionID, oldCdIDLocation)
+		if err != nil {
+			pair.Print(ctx, "stopping. Error occurred while approving cdid in the collection")
+			continue
+		}
+
```

Comment:
if this fails should we remove the added CDID from the collection, or is this very unlikely ?

Reply:
As discussed, If this fails it can be handled manually.